### PR TITLE
PR for #3531: create move-outline-to-first/last-child

### PR DESCRIPTION
**Pull request by edreamleo**
See #3531.

- [x] Create the commands.
- [x] Create new unit tests.
- [x] Extract commands create section definitions as the *first* child.

### CODE REVIEW

**Comment by edreamleo** [comment]
@boltex I hope you don't mind this PR. After all these years I realized how useful `move-outline-to-first-child` can be.

` test_move_outline_to_last_child` revealed a bug, so I think the unit tests are strong enough.

**Comment by boltex** [comment]
@edreamleo No problem at all, I'm adding it to leojs easily :)

**Comment by edreamleo** [comment]
@boltex Thanks. I really *do* think I'm done tweaking Leo, for now :-)

**Comment by boltex** [comment]
@edreamleo Do 'assert' and 'self.assert' actually perform the same thing? ( in unit tests, comparing those new tests to the other tests which use the unittest system's "self.assert" instead of normal "assert")

**Comment by boltex** [comment]
Note: I meant also to ask about using "self.assertEqual" which is used in all other tests instead of a normal 'assert'.

File: leo/unittests/commands/test_outlineCommands.py [hunk 1]
```diff
@@ -605,6 +605,65 @@ def test_sort_siblings(self):
...
+        assert first_child
+        first_gnx = first_child.gnx
+        # Move.
+        # c.selectPosition(first_child)
+        c.selectPosition(first_child)
+        c.moveOutlineToLastChild()
+        # Tests.
+        assert c.checkOutline() == 0
+        last_child = root.lastChild()
+        assert first_child
```
Comment (boltex) [inline leo/unittests/commands/test_outlineCommands.py:655]: Perhaps this was supposed to be an assertion of last_child?

File: leo/unittests/commands/test_outlineCommands.py [hunk 2]
```diff
@@ -605,6 +605,65 @@ def test_sort_siblings(self):
         u.undo()
         result_children = [z.h for z in self.root_p.v.children]
         self.assertEqual(result_children, original_children)
+    #@+node:ekr.20230902053728.1: *3* TestOutlineCommands.test_move_outline_to_first_child
+    def test_move_outline_to_first_child(self):
+        # Setup.
+        c, u = self.c, self.c.undoer
+        root = self.root_p
+        assert root.h == 'root'
```
Comment (boltex) [inline leo/unittests/commands/test_outlineCommands.py:613]: Should self.assertEqual be used ?

**Comment by edreamleo** [comment]
@boltex Good question about `assert` vs. `self.assertEqual`, etc.

The two are not equivalent, but the choice between the two typically matters only when either assert fails :-)

At first I thought perhaps unit tests should always use one or the other. Now I just use whatever is most convenient.

`self.assertEqual` can provide more helpful messages when it fails. Also, iirc, `self.assertX` do *not* report their failures when run with pytest: stdout gets sent to dev/null.

**Pull request merged**